src/layers/core/icon-layer/icon-layer-vertex.glsl.js

```diff
@@ -40,15 +41,22 @@ varying float vColorMode;
 varying vec4 vColor;
 varying vec2 vTextureCoords;
 
+vec2 rotate_by_angle(vec2 vertex, float angle) {
+  mat2 rotationMatrix = mat2(cos(angle), -sin(angle), sin(angle), cos(angle));
+  return rotationMatrix * vertex;
+}
+
 void main(void) {
   vec2 iconSize = instanceIconFrames.zw;
   vec2 iconSize_clipspace = iconSize / viewportSize * 2.0;
   // scale icon height to match instanceSize
   float instanceScale = iconSize.y == 0.0 ? 0.0 : instanceSizes / iconSize.y;
   vec3 center = project_position(instancePositions);
-  vec2 vertex = (positions / 2.0 + instanceOffsets) * iconSize_clipspace *
+  vec2 vertex = rotate_by_angle(positions, instanceAngles);
+  vertex = (vertex / 2.0 + instanceOffsets) * iconSize_clipspace *
```

Comment:
Icon should be rotated around the anchor point (specified by `instanceOffsets`). You can test this with the marker icon in layer-browser -- when zooming in and out, the base of the pin should remain at a fixed location.